server/services/caService.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * @typedef {import("../../types/delius").DeliusClient} DeliusClient
+ * @typedef {import("../../types/licences").CaService} CaService
+ */
+
+/**
+ * @param {DeliusClient} deliusClient
+ * @returns {CaService} caService
+ */
+
+const logger = require('../../log.js')
+const roService = require('./roService')
+
+module.exports = function createCaService(deliusClient, nomisClientBuilder) {
+  async function getLduForRo(bookingId, token) {
+    const ro = await roService(deliusClient, nomisClientBuilder).findResponsibleOfficer(bookingId, token)
+
+    if (ro.lduCode) {
+      logger.info(`Local delivery unit code received from Delius: ${ro.lduCode}`)
+    } else if (ro.message && ro.message.match(/Offender has not been assigned a COM:/)) {
+      logger.info('Offender has not been assigned a COM')
+    }
+
+    return ro.lduCode || ro.message
```

Comment:
there would be no way of being able to distinguish between these two results? They would both return a string - how you you know if its a valid ldu code?

Reply:
You'll also need to distinguish whether the RO is allocated or not: 

```
Call ROService.findResponsibleOfficerByBookingId
If error then return error
Call lduActiveClient with result.ldu_code 
Build a list of errors:
Error if not active
Error if !result.isAllocated
```

Reply:
ok